src/serial_bridge_index.cpp

```diff
@@ -634,20 +634,16 @@ string serial_bridge::decode_amount(int version, crypto::key_derivation derivati
 	if (version == 1) {
 		return amount;
 	} else if (version == 2) {
-		rct::key sk;
-		crypto::ec_scalar scalar = AUTO_VAL_INIT(scalar);
-		crypto::derivation_to_scalar(derivation, index, scalar);
-
-		string sk_str = epee::string_tools::pod_to_hex(scalar);
-		epee::string_tools::hex_to_pod(sk_str, sk);
+		crypto::secret_key scalar;
+		hw::get_device("default").derivation_to_scalar(derivation, index, scalar);
```

Comment:
Unified based on this https://github.com/monero-project/monero/blob/f47651578f55756823b8da3ebf9b28f752a5bc67/src/wallet/wallet2.cpp#L1769-L1794